src/transformers/tokenization_utils_base.py

```diff
@@ -1552,6 +1549,15 @@ def __init__(self, **kwargs):
         self._in_target_context_manager = False
         super().__init__(**kwargs)
 
+    @property
+    def added_tokens_encoder(self) -> Dict[str, int]:
+        # FIXME might not be very efficient, since not stored? Is len usually computed?
+        return {k.content: v for v, k in self._added_tokens_decoder.items()}
+
+    @property
+    def added_tokens_decoder(self) -> Dict[int, AddedToken]:
+        return self._added_tokens_decoder
```

Comment:
This removes user freedom, but is a lot more logical. Only one object to modify and can't have missmatches